polymer.externs.js

```diff
@@ -571,6 +571,46 @@ PolymerDomApi.prototype.insertBefore = function(node, beforeNode) {};
 /** @param {!Node} node */
 PolymerDomApi.prototype.removeChild = function(node) {};
 
+/**
+ * @typedef {function({
+ *   target: Node,
+ *   addedNodes: Array<Node>,
+ *   removedNodes: Array<Node>
+ * })}
+ */
+PolymerDomApi.ObserveCallback;
+
+/**
+ * A virtual type for observer callback handles.
+ *
+ * @private @constructor
+ */
+PolymerDomApi.ObserveHandle = function() {};
```

Comment:
I designed it this way so that it was a nominal type that the user can't create themselves.